### PR TITLE
[controller] Parent controller allows updating the current version in a special UpdateStore call

### PR DESCRIPTION
**Pull request by nisargthakkar**
## Parent controller allows updating the current version in a special UpdateStore call

In PR #862, we prevented parent controllers from updating the current version in Zk. However, we added a provision to allow the current version to still be updated by operators in a restricted mode. However, this custom upgrade doesn't work as it too gets skipped due to certain conditional checks. This commit fixes that issue.

## How was this PR tested?

GH CI

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 1]
```diff
@@ -4394,7 +4410,7 @@ private void internalUpdateStore(String clusterName, String storeName, UpdateSto
 
       if (currentVersion.isPresent()) {
         if (childRegionOnlyConfigUpdateAllowed) {
```
Comment (majisourav99) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:4412]: can we reuse the variable `onlyParentRegionFilter` ? Also simply the logic something  like

`if (currentVersion.isPresent()) {
        setStoreCurrentVersion(clusterName, storeName, currentVersion.get(), onlyParentRegionFilter);
 }`

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 2]
```diff
@@ -3718,9 +3718,22 @@ Pair<Store, Version> waitVersion(String clusterName, String storeName, int versi
...
+  private void setStoreCurrentVersion(
+      String clusterName,
+      String storeName,
+      int versionNumber,
+      boolean allowedInParent) {
     storeMetadataUpdate(clusterName, storeName, store -> {
-      if (isParent()) {
+      if (isParent() && !allowedInParent) {
         // Parent colo should not update the current version of a store
+        LOGGER.info("Updating current version in parent region is not allowed. Skipping.");
```
Comment (majisourav99) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:3736]: And add the logging here. Also never add logs without any identifying features, like store/cluster names.

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java (hunk 1 above)
Comment (nisargthakkar) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java]: Makes sense. Updated

**Review by majisourav99 — APPROVED** [review]
Thanks Nisarg

**Pull request merged**